shared/src/question-generators/time/pseudoCodeUtils.ts

```diff
@@ -0,0 +1,94 @@
+/*
+This package works, so we could use this for coloring the pseudocode
+https://www.w3schools.com/tags/ref_colornames.asp
+*/
+
+export type PseudoCodeString = Array<
+  string | PseudoCodeVariable | PseudoCodeFunctionName | PseudoCodePrintString
+>
+
+export type PseudoCodeVariable = { variable: string }
+export type PseudoCodeFunctionName = { functionName: string }
+export type PseudoCodePrintString = { printString: string }
+
+export type PseudoCodePrint = { print: PseudoCodeString }
+export type PseudoCodeCall = { functionName: string; args: Array<PseudoCodeString> }
+// return as it is a keyword in TS
+export type PseudoCodeReturn = { returnValue: PseudoCodeString }
+
+export type PseudoCodeBreak = { breakState: true }
+export type PseudoCodeContinue = { continueState: true }
+// export type PseudoCodeTrue = { true: true }
+// export type PseudoCodeFalse = { false: true }
+
+export type PseudoCodeAssignment = { variable: string; value: PseudoCodeString }
+
+export type PseudoCodeIf = {
+  if: {
+    condition: PseudoCodeString
+    then: PseudoCodeState | PseudoCodeBlock | null
+    elseif?: Array<{
```

Comment:
The `then` and the `else` part already allow for another `if` block. Maybe an explicit `elseif` block is not needed? but im not strongly against it.

Reply:
The `elseif` block is used to create inline `else if` instead of `else \n__if` and for converting the PseudoCode into Latex, because there is also a difference between `\Elsif` and `Else`.

Reply:
ok, then leave it as it is.